airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/DestinationInitialState.java

```diff
@@ -0,0 +1,22 @@
+package io.airbyte.integrations.base.destination.typing_deduping;
+
+import java.util.Optional;
+
+/**
+ * Interface representing the initial state of a destination table.
+ *
+ * @param <TableDefinition> type of the table definition.
+ */
+public interface DestinationInitialState<TableDefinition> {
+
+  StreamConfig streamConfig();
+
+  Optional<TableDefinition> finalTableDefinition();
+
+  InitialRawTableState initialRawTableState();
```

Comment:
any reason not to just promote the InitialRawTableState fields into this class + delete InitialRawTableState?